src/diff/props.js

```diff
@@ -50,7 +50,7 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 			if (typeof oldValue==='string') s.cssText = '';
 			// remove values not in the new list
 			for (let i in oldValue) {
-				if (value==null || !(i in value)) s.setProperty(i, '');
+				if (value==null || !(i in value)) s.setProperty(i.replace(/-?(?=[A-Z])/g, '-'), '');
```

Comment:
This looks like the same regex like a few lines below:

https://github.com/developit/preact/blob/4f671e7fb99425f84434c0c8b56d830ade4ea792/src/diff/props.js#L60

Maybe the size will be smaller, when we hoist them out into a variable outside of `setProperty()`. We usually compare the number at the top when running `npm run build:core` :+1:

Reply:
this solution is more smaller than outer function when gzipped.

Reply:
Are you sure? When hoisted the size is `5` bytes smaller. The PR currently sits at `3304 B` and when hoisted will be `3299 B`:

```js
let CAMEL_REG = /-?(?=[A-Z])/g;

function setProperty(dom, name, value, oldValue, isSvg) {
  //...
  if (value==null || !(i in value)) s.setProperty(i.replace(CAMEL_REG, '-'), '');

  // ...
  s.setProperty(i.replace(CAMEL_REG, '-'), typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v);
}
```

Reply:
oh, nice!
i tried this way:
```js
function hoge(elem, k, v){
  elem.setProperty(k.replace(/-?(?=[A-Z])/g, '-'), v);
}
```
ok, i'll do that.

Reply:
Ah I see, creating an additional function is indeed more costly than just hoisting out a variable :)